src/games/strategy/triplea/ai/proAI/ProData.java

```diff
@@ -1,43 +1,83 @@
 package games.strategy.triplea.ai.proAI;
 
 import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.PlayerID;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
 import games.strategy.triplea.Properties;
+import games.strategy.triplea.ai.proAI.data.ProPurchaseOption;
+import games.strategy.triplea.ai.proAI.data.ProPurchaseOptionMap;
+import games.strategy.triplea.ai.proAI.util.ProUtils;
+import games.strategy.triplea.attatchments.TerritoryAttachment;
+import games.strategy.triplea.delegate.BattleCalculator;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.util.IntegerMap;
+import games.strategy.util.Match;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
 /**
  * Pro AI data.
  */
 public class ProData {
 
+  private static ProAI proAI;
+  private static GameData data;
+  private static PlayerID player;
+
+  // Default values
   public static double winPercentage = 95;
   public static double minWinPercentage = 75;
   public static boolean areNeutralsPassableByAir = false;
+  public static Territory myCapital = null;
+  public static List<Territory> myUnitTerritories = new ArrayList<Territory>();
+  public static Map<Unit, Territory> unitTerritoryMap = new HashMap<Unit, Territory>();
+  public static IntegerMap<UnitType> unitValueMap = new IntegerMap<UnitType>();
+  public static ProPurchaseOptionMap purchaseOptions = null;
+  public static double minCostPerHitPoint = Double.MAX_VALUE;
 
-  private static GameData data;
-  private static ProAI proAI;
-
-  public static GameData getData() {
-    return data;
-  }
-
-  public static void setData(final GameData data) {
+  public static void initialize(final ProAI proAI, final GameData data, final PlayerID player) {
+    ProData.proAI = proAI;
```

Comment:
`public static` makes for a broad public API, so it's pretty reasonable to do null checks and check your parameters (fail early / fail fast principle).

Guava Preconditions would work well, can do a  static import of the "checkNotNull" method, so line 44 would be:
`ProData.proAi = checkNotNull(proAI)`